admiral/pkg/clusters/types.go

```diff
@@ -127,11 +130,185 @@ func (g *globalTrafficCache) Put(gtp *v1.GlobalTrafficPolicy) error {
+	filterCache map[string]map[string]map[string]string
+	mutex *sync.Mutex
+}
+
+func (r *routingPolicyFilterCache) Get(identityEnvKey string) (filters map[string]map[string]string) {
+	defer r.mutex.Unlock()
+	r.mutex.Lock()
+	return r.filterCache[identityEnvKey]
+}
+
+func (r *routingPolicyFilterCache) Put(identityEnvKey string, clusterId string, filterName string) {
+	defer r.mutex.Unlock()
+	r.mutex.Lock()
+	if r.filterCache[identityEnvKey] == nil {
+		r.filterCache[identityEnvKey] = make(map[string]map[string]string)
+	}
+
+	if r.filterCache[identityEnvKey][clusterId] == nil {
+		r.filterCache[identityEnvKey][clusterId] = make(map[string]string)
+	}
+	r.filterCache[identityEnvKey][clusterId][filterName] = filterName
+}
+
+func (r *routingPolicyFilterCache) Delete(identityEnvKey string) {
+	defer r.mutex.Unlock()
+	r.mutex.Lock()
+	// delete all envoyFilters for a given identity+env key
+	delete(r.filterCache, identityEnvKey)
+}
+func (r RoutingPolicyHandler) Added(obj *v1.RoutingPolicy) {
+	if common.ShouldIgnoreResource(obj.ObjectMeta) {
+		log.Infof(LogErrFormat, "success", "routingpolicy", obj.Name, obj.ClusterName, "Ignored the RoutingPolicy because of the annotation")
+		return
+	}
+	dependents, done := getDependents(obj, r)
+	if done {
+		return
+	}
+	r.processroutingPolicy(dependents, obj, admiral.Add)
+	log.Info("finished processing routing policy")
+}
+
+func (r RoutingPolicyHandler) processroutingPolicy(dependents map[string]string, routingPolicy *v1.RoutingPolicy, eventType admiral.EventType ) {
+	for _, remoteController := range r.RemoteRegistry.RemoteControllers {
+
+			for _, dependent := range dependents {
+
+			// Check if the dependent exists in this remoteCluster. If so, we create an envoyFilter with dependent identity as workload selector
+			if _, ok := r.RemoteRegistry.AdmiralCache.IdentityClusterCache.Get(dependent).Copy()[remoteController.ClusterID]; ok {
+
+				filter, err := createOrUpdateEnvoyFilter(remoteController, routingPolicy, eventType, dependent, r.RemoteRegistry.AdmiralCache)
+				if err != nil {
+					log.Errorf(LogErrFormat, admiral.Add, "routingpolicy", routingPolicy.Name, remoteController.ClusterID, err)
+				} else {
+					log.Infof("msg=%s name=%s cluster=%s", "created envoyfilter", filter.Name, remoteController.ClusterID)
+				}
+			}
+		}
+
+	}
+}
+
+func (r RoutingPolicyHandler) Updated(obj *v1.RoutingPolicy) {
+	if common.ShouldIgnoreResource(obj.ObjectMeta) {
+		log.Infof(LogFormat, admiral.Update, "routingpolicy", obj.Name, obj.ClusterName, "Ignored the RoutingPolicy because of the annotation")
+		// We need to process this as a delete event.
+		r.Deleted(obj)
+		return
+	}
+	dependents, missingIdentityLabel := getDependents(obj, r)
+	if missingIdentityLabel {
+		return
+	}
+	r.processroutingPolicy(dependents, obj, admiral.Update)
+	log.Info("updated routing policy")
+}
+
+func getDependents(obj *v1.RoutingPolicy, r RoutingPolicyHandler) (map[string]string, bool) {
```

Comment:
just a nitpick, the `bool` returned by the func is a little confusing, at least to me. Can we add a comment to this which explains when/why would the func return `true` vs `false`?

Reply:
removed it